src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,245 @@
+                revision=self.release,
+                token=self.hf_token,
+            )
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading GGUF model succeeded!",
+            fg="green",
+        )
+
+    def upload_safetensors(self) -> None:
+        try:
+            hf_api = HfApi()
+            hf_api.upload_folder(
+                folder_path=self.local_model_path,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+        except Exception as exc:
+            click.secho(
+                f"\nUploading safetensors model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading safetensors model succeeded!",
+            fg="green",
+        )
+
+
+@click.command()
+@click.option(
+    "--model",
+    "-m",
+    required=True,
+    help="Local model to upload.",
+)
+@click.option(
+    "--model-dir",
+    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
+    show_default="The default system model location store, located in the data directory.",
+    help="The local directory to upload the model files from.",
+)
+@click.option(
+    "--dest-type",
+    type=click.Choice(["hf", "oci", "s3"]),
+    required=True,
+    help="The type of destination to upload to - can be 'hf', 'oci', or 's3'",
+)
+@click.option(
+    "--upload-dest",
+    type=str,
+    required=True,
+    help="Destination for the model to be uploaded to. Ex: for a Hugging Face repo, should be '<username>/<repo-name>'",
+)
+@click.option(
+    "--release",
+    "-rl",
+    default="main",
+    show_default=True,
+    help="The revision to upload the model to - e.g. a branch for Hugging Face repositories.",
+)
+@click.option(
+    "--hf-token",
+    default="",
+    envvar="HF_TOKEN",
+    help="User access token for connecting to the Hugging Face Hub.",
+)
+@click.pass_context
+@clickext.display_params
+def upload(ctx, model, model_dir, dest_type, upload_dest, release, hf_token):
+    """Uploads model to a specified location"""
+    uploader = None
+
+    if dest_type == "hf":
+        uploader = HFModelUploader(
+            ctx=ctx,
+            model=model,
+            model_dir=model_dir,
+            upload_dest=upload_dest,
+            release=release,
+            hf_token=hf_token,
+        )
+    elif dest_type == "oci":
+        # TODO: update this when OCI uploading is supported
+        click.secho(
+            f"Uploading of type {dest_type} is not yet supported",
+            fg="yellow",
+        )
+        raise click.exceptions.Exit(0)
```

Comment:
Should this be exit 0?  Someone might think it worked.

Same question for the s3 case.

Reply:
My thought was that the command itself wasn't failing (expected output providing expected behavior) so 0 made more sense than 1 for an exit code

Reply:
I think of it as whether a shell script calling the command should consider it a success and keep going (with -e).  I would consider calling an upload that does nothing a failure.

Reply:
Fair point - I'm fine to change this